src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,319 @@
+            String key = enumerator.nextElement();
+            standardSession.setAttribute(key, attributes.get(key));
+        }
+
+        return standardSession;
+    }
+
+
+    /**
+     * Get metadata from a Session
+     * @param session
+     * @return
+     */
+    private Hashtable<String, Object> getMetadata(Session session){
+        Hashtable<String, Object> metadata = new Hashtable<>();
+        metadata.put(METADATA_VALID, String.valueOf(session.isValid()));
+        metadata.put(METADATA_CREATION_TIME, dateFormat.format(new Date(session.getCreationTime())));
+        metadata.put(METADATA_LAST_ACCESS_TIME, dateFormat.format(new Date(session.getLastAccessedTime())));
+        metadata.put(METADATA_MAX_INACTIVE_INTERVAL, String.valueOf(session.getMaxInactiveInterval()));
+        return metadata;
+    }
+
+    /**
+     * Get attributes from a Session
+     * @param session
+     * @return
+     */
+    private Hashtable<String, Object> getAttributes(Session session){
+        StandardSession standardSession = (StandardSession)session;
+        Hashtable<String, Object> attributes = new Hashtable();
+        for (Enumeration<String> enumerator = standardSession.getAttributeNames(); enumerator.hasMoreElements();) {
+            String key = enumerator.nextElement();
+            attributes.put(key, standardSession.getAttribute(key));
+        }
+        return attributes;
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // END Utils
+    //------------------------------------------------------------------------------------------------------------------
+
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START REDIS
+    //------------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Method to saved session in redis
+     * @param session
+     * @return
+     * @throws IOException
+     */
+    protected boolean saveSession(Session session) throws IOException {
+        boolean errorSaving = true;
+        //Get the session Id
+        String id = session.getId();
+        Hashtable<String, Object> metadata = getMetadata(session);
+        Hashtable<String, Object> attributes = getAttributes(session);
+        try{
+            //Save metadata to Redis
+            byte[] encodedMetadata = Base64.getEncoder().encode(SerializationUtils.serialize(metadata));
+            withRedis((Jedis jedis)-> jedis.set((id + REDIS_METADATA_KEY).getBytes(), encodedMetadata));
+
+            //Save Attributes to Redis
+            byte[] encodedAttributes = Base64.getEncoder().encode(SerializationUtils.serialize(attributes));
+            withRedis((Jedis jedis)-> jedis.setex((id + REDIS_ATTRIBUTES_KEY).getBytes(), session.getMaxInactiveInterval(), encodedAttributes));
```

Comment:
Make this call and the call on 224 into the same call, so you're only doing the pull from the pool once. You should also do these actions as a transaction within Redis, so that you don't end up with one update's attributes and another update's metadata.